### PR TITLE
[GH-96] Move the configuration into cardano-node.

### PR DESCRIPTION
**Pull request by ksaric**
https://github.com/input-output-hk/cardano-node/issues/96

The first step to moving the configuration from `cardano-shell` to `cardano-node`.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+test-ghci: ## Run repl on test suites
+	@stack ghci $(PROJECT_NAME):lib $(PROJECT_NAME):test:$(PROJECT_NAME)-test --ghci-options=-fobject-code --nix
+
+test-ghcid: ## Run ghcid on test suites
+	@ghcid --command "stack ghci --nix $(PROJECT_NAME):lib $(PROJECT_NAME):test:$(PROJECT_NAME)-test --ghci-options=-fobject-code"
+
+test-ghcid-nix: ## Run ghcid on test suites with Nix
+	@ghcid --command="stack ghci --test --main-is $(PROJECT_NAME):test:$(PROJECT_NAME)-test --nix -j$(NUM_PROC)"
+
+.PHONY: stylish-haskell ghcid ghcid-test run-test test-ghci test-ghcid help
```
Comment (deepfire) [inline Makefile]: I love the Makefile!

It always struck me as a particularly efficient way of providing canned scenarios in a repository!

Comment (ksaric) [reply Makefile]: This is from @HirotoShioi.

**Comment by deepfire** [comment]
This PR will help a lot to reduce friction between repositories!

One thing we should benefit from is changing the types for the parsers to be specific -- capitalising on https://github.com/input-output-hk/cardano-node/blob/master/cardano-node/Cardano/Node/CLI.hs -- but this is clearly out of the scope of this PR.

**Pull request merged**